src/ts_catalog/continuous_agg.h

```diff
@@ -65,8 +65,8 @@ typedef struct ContinuousAggsBucketFunction
 	char *name;
 	/* `bucket_width` argument of the function */
 	Interval *bucket_width;
-	/* `origin` argument of the function provided by the user */
-	char *origin;
+	/* `origin` argument of the function. If not specified, stores infinity */
+	Timestamp origin;
```

Comment:
Shouldn't `origin` be a `union` (or something else) and include `TimestampTz` as well? The origin is not always `Timestamp`. At the moment it's the same implementation, but that might change in the future and it's not very clear in the code that this might be a `TimestampTz` after all.

Also, what about `ts_time_bucket_ng_date()`? Don't they also have origin? Do we use those?

Reply:
Timestamp and TimestampTz are exactly the same in PostgreSQL, thus we cat get away with this. I am OK with using union's but I think this will make the code unnecessarily complicated and a bit slower (because we end up checking if timestamp or timestamptz is stored). Although I agree there should be a comment about this. I will add one.

DATEs can be converted to TIMESTAMPs and vice versa without any loss of data. I will add a comment about this as well.

Reply:
Hopefully, it's better now. Feel free to re-open this discussion or create a new one if necessary.